frontend/src/metabase/browse/containers/BrowseModels.tsx

```diff
@@ -0,0 +1,278 @@
+  useEffect(() => {
+    const configureGrid = () => {
+      const gridOptions = getGridOptions(models);
+      setGridOptions(gridOptions);
+    };
+    configureGrid();
+    window.addEventListener("resize", configureGrid);
+    return () => window.removeEventListener("resize", configureGrid);
+  }, [models]);
+
+  const [gridOptions, setGridOptions] = useState<{
+    cells: Cell[];
+  } | null>(null);
+
+  if (error) {
+    return <LoadingAndErrorWrapper error />;
+  } else if (isLoading || !gridOptions) {
+    return (
+      <LoadingAndErrorWrapper loading style={{ display: "flex", flex: 1 }} />
+    );
+  }
+
+  const { cells = [] } = gridOptions;
+
+  return cells.length ? (
+    <GridContainer role="grid">{cells}</GridContainer>
+  ) : (
+    <CenteredEmptyState
+      title={t`No models here yet`}
+      message={t`Models help curate data to make it easier to find answers to questions all in one place.`}
+      illustrationElement={<img src={NoResults} />}
+    />
+  );
+};
+
+interface ModelCellProps {
+  model: SearchResult;
+  style?: React.CSSProperties;
+  collectionHtmlId: string;
+}
+
+const ModelCell = ({ model, style, collectionHtmlId }: ModelCellProps) => {
+  const modelWithHistory = addLastEditInfo(model);
+  const lastEdit = modelWithHistory["last-edit-info"];
+  const lastEditorName = lastEdit.full_name;
+  const howLongAgo = getHowLongAgo(lastEdit.timestamp);
+
+  const headingId = `heading-for-model-${model.id}`;
+
+  const formattedDate = formatDateTimeWithUnit(lastEdit.timestamp, "day", {});
+
+  const time = <time dateTime={lastEdit.timestamp}>{formattedDate}</time>;
+  const tooltipLabel = jt`Last edited by ${lastEditorName}${(<br />)}${time}`;
+
+  return (
+    <Link
+      aria-labelledby={`${collectionHtmlId} ${headingId}`}
+      key={model.id}
+      style={style}
+      to={Urls.model(model as unknown as Partial<Card>)}
+      // FIXME: Not sure that 'Model Click' is right; this is modeled on the database grid which has 'Database Click'
+      data-metabase-event={`${ANALYTICS_CONTEXT};Model Click`}
+    >
+      <ModelCard>
+        <h4 className="text-wrap" style={{ lineHeight: "16px" }}>
+          <MultilineEllipsified id={headingId}>
+            {model.name}
+          </MultilineEllipsified>
+        </h4>
+        <Text size="xs" style={{ height: "32px" }}>
+          <MultilineEllipsified
+            tooltipMaxWidth="100%"
+            className={model.description ? "" : "text-light"}
+          >
+            {model.description || "No description."}{" "}
+          </MultilineEllipsified>
+        </Text>
+        <LastEditInfoLabel
+          item={modelWithHistory}
+          fullName={modelWithHistory["last-edit-info"].full_name}
+          className={"last-edit-info-label-button"}
+          tooltipProps={{
+            label: tooltipLabel,
+            withArrow: true,
+          }}
+          // TODO: Use first name and last initial
```

Comment:
There is an open question about whether the "Connor D" format makes sense, so for now we are using the last editor's full name.